### PR TITLE
Standardize event data for accumulators’ data

### PR DESCRIPTION
**Pull request by Gerson2102**
Adding the functionality to standardize. Resolves this #220.

### CODE REVIEW

File: data_handler/handler_tools/data_parser/serializers.py [hunk 1]
```diff
@@ -1 +1,9 @@
+from decimal import Decimal
 
+from pydantic import BaseModel
+
+
+class AccumulatorsSyncEvent(BaseModel):
```
[the remark this answers is not in the document]
Comment (Gerson2102) [reply data_handler/handler_tools/data_parser/serializers.py:6]: Got it, can you give an example? Maybe DataSyncEvent?

File: data_handler/handlers/loan_states/zklend/events.py [hunk 2]
```diff
@@ -161,13 +162,11 @@ def process_accumulators_sync_event(self, event: pd.Series) -> None:
...
         # Example: https://starkscan.co/event/0x029628b89875a98c1c64ae206e7eb65669cb478a24449f3485f5e98aba6204dc_0.
         # TODO: Integrate the ZEND token once it's allowed to be borrowed or used as collateral.
-        token = add_leading_zeros(event["data"][0])
-        collateral_interest_rate_index = decimal.Decimal(
-            str(int(event["data"][1], base=16))
-        ) / decimal.Decimal("1e27")
-        debt_interest_rate_index = decimal.Decimal(
-            str(int(event["data"][2], base=16))
-        ) / decimal.Decimal("1e27")
+        parsed_event = ZklendDataParser.parse_accumulators_sync_event(event["data"])
```
[the remark this answers is not in the document]
Comment (Gerson2102) [reply data_handler/handlers/loan_states/zklend/events.py:165]: I dont get this, can you elaborate more?

File: data_handler/handler_tools/data_parser/zklend.py [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
     Parses the zkLend data to human-readable format.
     """
 
     @classmethod
     def parse_accumulators_sync_event(cls, event_data):
-        # TODO: Implement parsing logic for AccumulatorsSync event
-        pass
+        parsed_event = AccumulatorsSyncEvent(
+            token=event_data[0],
+            lending_accumulator=Decimal(int(event_data[1], 16)) / Decimal("1e27"),
```
[the remark this answers is not in the document]
Comment (djeck1432) [reply data_handler/handler_tools/data_parser/zklend.py:15]: Here is docs about validator in pydantic
https://docs.pydantic.dev/latest/concepts/validators/#annotated-validators

this part should be in field_validator in your pydantic models
`Decimal(int(event_data[1], 16)) / Decimal("1e27"),`
so you will do next:
```
AccumulatorsSyncEvent(lending_accumulator=event_data[1])
and in `AccumulatorsSyncEvent.validate_fields` will be the logic of it `Decimal(int(event_data[1], 16)) / Decimal("1e27"),`

p.s. it's example of code

File: data_handler/handlers/loan_states/zklend/events.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (djeck1432) [reply data_handler/handlers/loan_states/zklend/events.py:165]: add to the end of the variable name, `_data`
`parsed_event` + `_data`

**Comment by djeck1432** [comment]
@Gerson2102 Cool, will check it tomorrow

File: data_handler/handler_tools/data_parser/serializers.py [hunk 4]
```diff
@@ -1 +1,13 @@
...
+from pydantic import BaseModel, field_validator
+
+
+class DataAccumulatorsSyncEvent(BaseModel):
+    token: str
+    lending_accumulator: Decimal
+    debt_accumulator: Decimal
+
+    @field_validator("lending_accumulator", "debt_accumulator", mode="before")
+    def hex_to_decimal(cls, v):
```
Comment (djeck1432) [inline data_handler/handler_tools/data_parser/serializers.py:12]: add docstring and annotation type and return type

**Review by djeck1432 — CHANGES_REQUESTED** [review]

**Comment by Gerson2102** [comment]
Done here mate @djeck1432

**Review by djeck1432 — APPROVED** [review]
Looks good to me

**Pull request merged**